crates/sui-bridge/src/lib.rs

```diff
@@ -28,13 +28,65 @@ pub(crate) mod sui_mock_client;
 #[cfg(test)]
 pub(crate) mod test_utils;
 
-// TODO: can we log the error very time it gets retried?
 #[macro_export]
-macro_rules! retry_with_max_delay {
-    ($func:expr, $max_delay:expr) => {{
-        let retry_strategy = ExponentialBackoff::from_millis(100)
-            .max_delay($max_delay)
-            .map(jitter);
-        Retry::spawn(retry_strategy, || $func).await
+macro_rules! retry_with_max_elapsed_time {
```

Comment:
I didn't realize earlier that timeout is desirable for this macro. Why not wrap the retries with `tokio::time::timeout()`? Also at a glance it does look safe to not retry forever for the current callsites, but it would be nice to make sure, e.g. no event would be missed due to transient failures.

Reply:
> make sure, e.g. no event would be missed due to transient failures.

this is guaranteed by not moving the watermark until the events are done being processed